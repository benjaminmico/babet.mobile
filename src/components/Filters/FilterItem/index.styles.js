import styled from 'styled-components'

export const FilterItemContainer = styled.TouchableOpacity`
  background-color: ${props => props.backgroundColor};
  border-radius: 8px;
  margin-top: ${props => (props.index === 0 ? '0px' : '11px')};
  width: 74px;
  height: 24px;
  justify-content: center;
  align-items: center;
`

export const FilterItemText = styled.Text`
  color: ${props => props.color};
  font-family: 'AvenirNext-DemiBold';
  font-size: 13px;
  line-height: 18px;
  letter-spacing: -0.14px;
`
