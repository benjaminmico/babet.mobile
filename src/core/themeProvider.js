import AsyncStorage from '@react-native-community/async-storage'
import React, {useContext, useEffect, useState} from 'react'
import {useDarkMode} from 'react-native-dark-mode'
import THEMES from './themes.json'

const STORAGE_KEY = 'THEME_ID'
const ThemeContext = React.createContext()

export const ThemeContextProvider = ({children}) => {
  const [themeID, setThemeID] = useState()

  const isDarkMode = useDarkMode ? useDarkMode() : false

  useEffect(() => {
    ;(async () => {
      // eslint-disable-next-line no-unused-expressions
      isDarkMode ? setThemeID(THEMES[0].key) : setThemeID(THEMES[1].key)
      // const storedThemeID = await AsyncStorage.getItem(STORAGE_KEY)
      // if (storedThemeID) setThemeID(storedThemeID)
      // // eslint-disable-next-line no-unused-expressions
      // else isDarkMode ? setThemeID(THEMES[0].key) : setThemeID(THEMES[1].key)
    })()
  }, [])

  return <ThemeContext.Provider value={{themeID, setThemeID}}>{themeID ? children : null}</ThemeContext.Provider>
}

export function withTheme(Component) {
  return props => {
    const {themeID, setThemeID} = useContext(ThemeContext)

    const getTheme = themeID => THEMES.find(theme => theme.key === themeID)
    const setTheme = themeID => {
      AsyncStorage.setItem(STORAGE_KEY, themeID)
      setThemeID(themeID)
    }

    return <Component {...props} themes={THEMES} theme={getTheme(themeID)} setTheme={setTheme} />
  }
}

export function getTheme() {
  const {themeID} = useContext(ThemeContext)

  const getThemeContent = themeID => THEMES.find(theme => theme.key === themeID)

  return getThemeContent(themeID)
}
