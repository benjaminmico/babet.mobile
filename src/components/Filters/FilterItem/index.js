// @flow

import {withTheme} from '@core/themeProvider'
import React from 'react'
// import {useTranslation} from 'react-i18next'
import {FilterItemContainer, FilterItemText} from './index.styles'

type Props = {
  // index item position
  index: Number,
  // filter label
  label: String,
  // check if filter item is selected or not
  selected: Boolean,
  // theme
  theme: Object,
}

const FilterItem = ({index, label, selected, theme}: Props) => {
  //   const {t} = useTranslation()

  // get theme props
  const {
    colors: {
      components: {backgroundFilter},
      texts: {
        filters: {selected: colorSelected, unselected: colorUnselected},
      },
    },
  } = theme

  return (
    <FilterItemContainer
      testID="FILTER_ITEM_CONTAINER"
      index={index}
      backgroundColor={selected ? backgroundFilter : 'transparent'}
    >
      <FilterItemText color={selected ? colorSelected : colorUnselected}>{label}</FilterItemText>
    </FilterItemContainer>
  )
}

export default withTheme(FilterItem)
