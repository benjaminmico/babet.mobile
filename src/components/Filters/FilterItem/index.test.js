/**
 * @jest-environment jsdom
 */
import React from 'react'
import {StyleSheet} from 'react-native'
import {ThemeContextProvider} from '@core/themeProvider'
import {shallow, mount} from 'enzyme'
import FilterItem from '.'

describe.only('./FilterItem', () => {
  describe('Integrity tests', () => {
    it('renders FilterItem correctly', () => {
      const wrapper = shallow(
        <ThemeContextProvider>
          <FilterItem index={0} label="aaaaa" />
        </ThemeContextProvider>,
      )

      expect(wrapper).toMatchSnapshot()
    })
    // it('renders InputAction with Left Button correctly', () => {
    //   const wrapper = shallow(<InputAction leftButton />)

    //   const LeftButton = wrapper.findWhere(node => node.prop('testID') === 'LEFT_BUTTON')

    //   expect(LeftButton.exists()).toBe(true)
    // })

    // it('renders InputAction without Left Button correctly', () => {
    //   const wrapper = shallow(<InputAction />)

    //   const LeftButton = wrapper.findWhere(node => node.prop('testID') === 'LEFT_BUTTON')

    //   expect(LeftButton.exists()).toBe(false)
    // })

    // it('renders InputAction with a placeholder', () => {
    //   const wrapper = shallow(<InputAction leftButton placeholder="Enter your message" />)

    //   const InputComponent = wrapper.findWhere(node => node.prop('testID') === 'INPUT_COMPONENT')

    //   expect(InputComponent.prop('placeholder')).toBe('Enter your message')
    // })

    // it('renders InputAction with Button Container correctly', () => {
    //   const wrapper = shallow(<InputAction leftButton />)

    //   const ButtonContainer = wrapper.findWhere(node => node.prop('testID') === 'BUTTON_CONTAINER')

    //   expect(ButtonContainer.exists()).toBe(true)
    // })

    // it('renders InputAction with a predefined value correctly', () => {
    //   const wrapper = shallow(<InputAction value="This is my message" />)

    //   const InputComponent = wrapper.findWhere(node => node.prop('testID') === 'INPUT_COMPONENT')

    //   expect(InputComponent.prop('value')).toBe('This is my message')
    // })
  })

  describe('Props tests', () => {
    it('renders selected FilterItem', () => {
      const wrapper = mount(
        <ThemeContextProvider>
          <FilterItem index={0} label="aaaaa" />
        </ThemeContextProvider>,
      )

      const FilterItemContainer = wrapper.find('TouchableOpacity').first()

      expect(wrapper).toMatchSnapshot()

      const styles = StyleSheet.flatten(FilterItemContainer.prop('style'))
      expect(styles.backgroundColor).toBe('transparent')
      expect(styles.borderRadius).toBe(8)
      expect(styles.marginTop).toBe(0)
      expect(styles.width).toBe(74)
      expect(styles.height).toBe(24)
      expect(styles.justifyContent).toBe('center')
      expect(styles.alignItems).toBe('center')
      //   expect(componentStyles.alignItems).toBe('center')
      //   expect(componentStyles.width).toBe(DIMENSIONS_WIDTH)
      //   expect(componentStyles.backgroundColor).toBe('white')
      //   expect(componentStyles.paddingBottom).toBe(0)
      // })
    })
  })
})
