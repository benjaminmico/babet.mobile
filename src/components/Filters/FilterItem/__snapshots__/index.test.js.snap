// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`./FilterItem Integrity tests renders FilterItem correctly 1`] = `
<ContextProvider
  value={
    Object {
      "setThemeID": [Function],
      "themeID": undefined,
    }
  }
/>
`;

exports[`./FilterItem Props tests renders selected FilterItem 1`] = `
<ThemeContextProvider>
  <Component
    index={0}
    label="aaaaa"
  >
    <FilterItem
      index={0}
      label="aaaaa"
      setTheme={[Function]}
      theme={
        Object {
          "backgroundColor": "#ffffff",
          "color": "#101112",
          "colors": Object {
            "components": Object {
              "background": "#FFFFFF",
              "backgroundAction": "#5100FF",
              "backgroundContainer": "#FFFFFF",
              "backgroundFilter": "#F3F2F2",
              "icon": "#000000",
            },
            "palette": Object {
              "black": "#000000",
              "purple": "#5100FF",
              "white": "#FFFFFF",
            },
            "texts": Object {
              "buttons": Object {
                "actions": "#5100FF",
                "filter": "#3E3E3E",
              },
              "description": "#ADADAD",
              "filters": Object {
                "selected": "#000000",
                "unselected": "#A3A2A2",
              },
              "labelInput": "#7C7C7C",
              "text": "#000000",
              "title": "#000000",
            },
          },
          "key": "white",
          "statusBar": "dark-content",
        }
      }
      themes={
        Array [
          Object {
            "backgroundColor": "#101112",
            "color": "#ffffff",
            "colors": Object {
              "components": Object {
                "background": "#101112",
                "backgroundAction": "#5100FF",
                "backgroundContainer": "#2F3134",
                "backgroundFilter": "#F3F2F2",
                "icon": "#FFFFFF",
              },
              "palette": Object {
                "black": "#000000",
                "purple": "#5100FF",
                "white": "#FFFFFF",
              },
              "texts": Object {
                "buttons": Object {
                  "actions": "#5100FF",
                  "filter": "#3E3E3E",
                },
                "description": "#ADADAD",
                "filters": Object {
                  "selected": "#000000",
                  "unselected": "#A3A2A2",
                },
                "labelInput": "#7C7C7C",
                "text": "#FFFFFF",
                "title": "#000000",
              },
            },
            "key": "dark",
            "statusBar": "light-content",
          },
          Object {
            "backgroundColor": "#ffffff",
            "color": "#101112",
            "colors": Object {
              "components": Object {
                "background": "#FFFFFF",
                "backgroundAction": "#5100FF",
                "backgroundContainer": "#FFFFFF",
                "backgroundFilter": "#F3F2F2",
                "icon": "#000000",
              },
              "palette": Object {
                "black": "#000000",
                "purple": "#5100FF",
                "white": "#FFFFFF",
              },
              "texts": Object {
                "buttons": Object {
                  "actions": "#5100FF",
                  "filter": "#3E3E3E",
                },
                "description": "#ADADAD",
                "filters": Object {
                  "selected": "#000000",
                  "unselected": "#A3A2A2",
                },
                "labelInput": "#7C7C7C",
                "text": "#000000",
                "title": "#000000",
              },
            },
            "key": "white",
            "statusBar": "dark-content",
          },
        ]
      }
    >
      <Styled(Component)
        backgroundColor="transparent"
        index={0}
        testID="FILTER_ITEM_CONTAINER"
      >
        <StyledNativeComponent
          backgroundColor="transparent"
          forwardedComponent={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": Array [],
              "displayName": "Styled(Component)",
              "inlineStyle": InlineStyle {
                "rules": Array [
                  "
  background-color: ",
                  [Function],
                  ";
  border-radius: 8px;
  margin-top: ",
                  [Function],
                  ";
  width: 74px;
  height: 24px;
  justify-content: center;
  align-items: center;
",
                ],
              },
              "render": [Function],
              "styledComponentId": "StyledNativeComponent",
              "target": Object {
                "$$typeof": Symbol(react.forward_ref),
                "render": [Function],
              },
              "withComponent": [Function],
            }
          }
          forwardedRef={null}
          index={0}
          testID="FILTER_ITEM_CONTAINER"
        >
          <ForwardRef
            backgroundColor="transparent"
            index={0}
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "borderRadius": 8,
                  "height": 24,
                  "justifyContent": "center",
                  "marginTop": 0,
                  "width": 74,
                },
              ]
            }
            testID="FILTER_ITEM_CONTAINER"
          >
            <TouchableOpacity
              backgroundColor="transparent"
              hostRef={null}
              index={0}
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "transparent",
                    "borderRadius": 8,
                    "height": 24,
                    "justifyContent": "center",
                    "marginTop": 0,
                    "width": 74,
                  },
                ]
              }
              testID="FILTER_ITEM_CONTAINER"
            >
              <ForwardRef(AnimatedComponentWrapper)
                accessible={true}
                focusable={false}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Array [
                    Array [
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "transparent",
                        "borderRadius": 8,
                        "height": 24,
                        "justifyContent": "center",
                        "marginTop": 0,
                        "width": 74,
                      },
                    ],
                    Object {
                      "opacity": 1,
                    },
                  ]
                }
                testID="FILTER_ITEM_CONTAINER"
              >
                <AnimatedComponent
                  accessible={true}
                  focusable={false}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Array [
                      Array [
                        Object {
                          "alignItems": "center",
                          "backgroundColor": "transparent",
                          "borderRadius": 8,
                          "height": 24,
                          "justifyContent": "center",
                          "marginTop": 0,
                          "width": 74,
                        },
                      ],
                      Object {
                        "opacity": 1,
                      },
                    ]
                  }
                  testID="FILTER_ITEM_CONTAINER"
                >
                  <View
                    accessible={true}
                    focusable={false}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "transparent",
                        "borderRadius": 8,
                        "height": 24,
                        "justifyContent": "center",
                        "marginTop": 0,
                        "opacity": 1,
                        "width": 74,
                      }
                    }
                    testID="FILTER_ITEM_CONTAINER"
                  >
                    <View
                      accessible={true}
                      focusable={false}
                      onClick={[Function]}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        Object {
                          "alignItems": "center",
                          "backgroundColor": "transparent",
                          "borderRadius": 8,
                          "height": 24,
                          "justifyContent": "center",
                          "marginTop": 0,
                          "opacity": 1,
                          "width": 74,
                        }
                      }
                      testID="FILTER_ITEM_CONTAINER"
                    >
                      <Styled(Text)
                        color="#A3A2A2"
                      >
                        <StyledNativeComponent
                          color="#A3A2A2"
                          forwardedComponent={
                            Object {
                              "$$typeof": Symbol(react.forward_ref),
                              "attrs": Array [],
                              "displayName": "Styled(Text)",
                              "inlineStyle": InlineStyle {
                                "rules": Array [
                                  "
  color: ",
                                  [Function],
                                  ";
  font-family: 'AvenirNext-DemiBold';
  font-size: 13px;
  line-height: 18px;
  letter-spacing: -0.14px;
",
                                ],
                              },
                              "render": [Function],
                              "styledComponentId": "StyledNativeComponent",
                              "target": [Function],
                              "withComponent": [Function],
                            }
                          }
                          forwardedRef={null}
                        >
                          <Text
                            color="#A3A2A2"
                            style={
                              Array [
                                Object {
                                  "color": "#A3A2A2",
                                  "fontFamily": "AvenirNext-DemiBold",
                                  "fontSize": 13,
                                  "letterSpacing": -0.14,
                                  "lineHeight": 18,
                                },
                              ]
                            }
                          >
                            <Text
                              color="#A3A2A2"
                              style={
                                Array [
                                  Object {
                                    "color": "#A3A2A2",
                                    "fontFamily": "AvenirNext-DemiBold",
                                    "fontSize": 13,
                                    "letterSpacing": -0.14,
                                    "lineHeight": 18,
                                  },
                                ]
                              }
                            >
                              aaaaa
                            </Text>
                          </Text>
                        </StyledNativeComponent>
                      </Styled(Text)>
                      <PressabilityDebugView
                        color="cyan"
                      />
                    </View>
                  </View>
                </AnimatedComponent>
              </ForwardRef(AnimatedComponentWrapper)>
            </TouchableOpacity>
          </ForwardRef>
        </StyledNativeComponent>
      </Styled(Component)>
    </FilterItem>
  </Component>
</ThemeContextProvider>
`;
